### DISCUSSION TITLE
What's the easiest way to create a blocking activity?

### DISCUSSION THREAD

**Discussion by mrubelmann** [elsa-workflows/elsa-core#2685] [Q&A]
I've got a custom async activity that I would like to turn into a blocking activity.  I've looked at the instructions on how to do it, but despite how incredibly thorough it is, I just can't wrap my brain around it.  Do we really need the bookmarks and stuff?  There's no `Blocking = true` flag somewhere?  Is there a simpler way to do it?  I'm using this within a composite activity, so can I somehow use a built-in activity to make part of it blocking?  Maybe signals or `UserTask`?  It would be awesome if I could do something like `builder.ThenBlocking<MyActivity>().WriteLine("All done");` or `builder.Then<MyActivity>().JustHangOutHereUntilTheLastThingFinishes().WriteLine("All done");`

**Comment by sfmskywalker** [comment]
Yeah I hear ya, it's indeed no trivial story. The easiest way to create a blocking activity is to write a custom activity that returns a `SuspendResult` - that's enough for the workflow to become suspended & blocked on that activity.

The bookmarks are necessary only for finding out what workflow instances can be resumed for a given set of inputs. But if you don't need any of that, you can do without. All you need to resume a suspended workflow are the following bits of information:

- Workflow instance ID **or** correlation ID.
- The activity ID to resume.

You can use the `IResumesWorkflow` to manually resume a given workflow instance.

Your suggestion to be able to do e.g. `builder.ThenBlocking<MyActivity>` is an interesting one that I'll give some more thought.
